libs/novui/src/json-schema-components/utils.ts

```diff
@@ -2,6 +2,19 @@ import { JsonSchemaFormSectionVariant } from '../../styled-system/recipes';
 
 export const JSON_SCHEMA_FORM_ID_DELIMITER = '~~~';
 
+export const AUTOCOMPLETE_OPEN_TAG = '{{';
+export const AUTOCOMPLETE_CLOSE_TAG = '}}';
+
+export const AUTOCOMPLETE_REGEX = new RegExp(`${AUTOCOMPLETE_OPEN_TAG}(.*?(.*?))${AUTOCOMPLETE_CLOSE_TAG}`, 'm');
+
+export const VALID_VARIABLE_REGEX = new RegExp(`${AUTOCOMPLETE_OPEN_TAG}([^{}]*?)${AUTOCOMPLETE_CLOSE_TAG}`, 'g');
+
+// To get initial invalid syntax of {{..} or {{..<space>
+export const INVALID_VARIABLE_REGEX = new RegExp(
+  `(${AUTOCOMPLETE_OPEN_TAG}[^{}|\\s]*[}|\\s](?!})|{{[^{}]*}(?!}})|{{[^{}]*$)`,
+  'g'
+);
+
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** I believe I made a mistake by putting `JSON_SCHEMA_FORM_ID_DELIMITER` here -- could we please move that and all these RegEx-related values to a separate `constants` file?‏